Lib/xml/etree/ElementTree.py

```diff
@@ -1200,6 +1200,12 @@ def parse(source, parser=None):
     Return an ElementTree instance.
 
     """
+    if getattr(source, 'encoding', None) == 'ISO-8859-1' and source.mode == 'r':
+        import warnings
+        warnings.warn(
+            "For file objects containing XML data "
+            "with non-ASCII and non-UTF-8 encoding (for example. ISO 8859-1), "
+            "the file must have been opened in binary mode.", category=RuntimeWarning)
```

Comment:
Honestly, I don't understand this warning message :-( I am not sure how to rephase it.

The warning is emitted even if XML is encoded to ASCII. "Non-UTF-8" is unclear to me. I'm not sure why ISO 8859-1 is only given as an example, whereas the warning is only emitted if the encoding is ISO-8859-1.

Should we also emit the warning for ISO-8859-15? What about ShiftJIS? Or Big5?

Let me try to propose a better message.

"File using {source.encoding} encoding should open XML in binary mode."

And consider emitting the warning if the encoding is not ASCII nor UTF-8.

I'm not a XML expert, so I'm not sure :-(

Reply:
In fact, `non-UTF8` and `non-ASCII` refer to something similar to **ISO-8859-1**, But in fact the character set it contains is not similar to `ASCII`, and it isn't considered `UTF8`. I agree with your point of view, it should be shortened, I think it should be changed to `ISO 8859-1 encoding should be read using 'rb' mode`.

Reply:
> Should we also emit the warning for ISO-8859-15? What about ShiftJIS? Or Big5?

IMO, that should be identified (on the issue) first. It seems weird to only add a warning for one particular non-UTF8 encoding. I would guess that the problem is not limited to ISO-8859-1.

Reply:
Why only 8-bit encodings? What about UTF-16? Or any other multibyte encoding?

Reply:
Because this occurs in 8-bit encoding and has been read in non-rb mode (although I can't guarantee that this will not happen in 16-bit, in fact, 16-bit encoding can be read with r instead of rb)

Reply:
I convert PR into a draft and continue to track the problem.